provider/ec2/environ.go

```diff
@@ -411,6 +425,12 @@ func (e *environ) StartInstance(args environs.StartInstanceParams) (_ *environs.
 		if placement.availabilityZone.State != availableState {
 			return nil, errors.Errorf("availability zone %q is %q", placement.availabilityZone.Name, placement.availabilityZone.State)
 		}
+		if volumeAttachmentsZone != "" && volumeAttachmentsZone != placement.availabilityZone.Name {
+			return nil, errors.Errorf(
+				"placement %q conflicts with zone %q for volume attachment(s)",
```

Comment:
Happy to say EBS volumes instead of volume attachments. The message already includes the zone of the volumes: "... conflicts with **zone "%q" for volume attachment(s)**"